src/pymor/algorithms/timestepping.py

```diff
@@ -10,227 +10,619 @@
+
+        self.step_function = step_function
+
+    def _step_function(self, U_n, t_n):
+        return self.step_function(U_n, t_n)
+
 
 class ImplicitEulerTimeStepper(TimeStepper):
-    """Implicit Euler time-stepper.
+    """Implicit Euler :class:`TimeStepper`.
 
     Solves equations of the form ::
 
-        M * d_t u + A(u, mu, t) = F(mu, t).
+        M * d_t u + A(u, mu, t) = F(mu, t),
+                     u(mu, t_0) = u_0(mu),
+
+    by an implicit Euler time integration, implemented in :class:`ImplicitEulerIterator`.
 
     Parameters
     ----------
     nt
         The number of time-steps the time-stepper will perform.
+    num_values
+        The number of returned vectors of the solution trajectory. If `None`, each intermediate vector that is
+        calculated is returned. Else an interpolation of the calculated vectors on an equidistant temporal grid is
+        returned, using an appropriate interpolation of the respective time stepper.
     solver_options
-        The |solver_options| used to invert `M + dt*A`.
-        The special values `'mass'` and `'operator'` are
-        recognized, in which case the solver_options of
-        M (resp. A) are used.
+        The |solver_options| used to invert `M + dt*A`. The special values `'mass'` and `'operator'` are recognized,
+        in which case the solver_options of `M` (resp. `A`) are used.
+    interpolation
+        Type of temporal interpolation to be used. Currently implemented are: piecewise constant (P0) and piecewise
+        linear (P1).
     """
 
-    def __init__(self, nt, solver_options='operator'):
+    IteratorType = ImplicitEulerIterator
+
+    def __init__(self, nt, num_values=None, solver_options='operator', interpolation='P1'):
+        super().__init__(num_values, interpolation)
+
+        assert isinstance(nt, Number)
+        assert nt > 0
+
         self.__auto_init(locals())
 
-    def solve(self, initial_time, end_time, initial_data, operator, rhs=None, mass=None, mu=None, num_values=None):
-        return implicit_euler(operator, rhs, mass, initial_data, initial_time, end_time, self.nt, mu, num_values,
-                              solver_options=self.solver_options)
+
+class ExplicitEulerIterator(SingleStepTimeStepperIterator):
+
+    def __init__(self, stepper, t0, t1, U0, A, F, M, mu, return_iter, return_times):
+        super().__init__(stepper, t0, t1, U0, A, F, M, mu, return_iter, return_times)
+        self.dt = dt = (t1 - t0) / stepper.nt
+        # use the ones from base, these are checked and converted in super().__init__()
+        A, F, M, mu = self.A, self.F, self.M, self.mu
+
+        # prepare the step function U_np1 = M^{-1}(M U_n + dt F - dt A U_n)
+        if not 't' in M.parameters:
+            M = M.assemble(mu)
+        if not 't' in A.parameters:
+            A = A.assemble(mu)
+
+        if isinstance(F, ZeroOperator):
+            if isinstance(M, IdentityOperator):
+                def step_function(U_n, t_n):
```

Comment:
Same comment as above.

Reply:
Same answer as before. I did timings back then, which I showed you, where you decided we want to have this branching here. Do you want to reconsider?

Reply:
I see. Do you still have those timings? I find that quite surprising.